pandas/core/arrays/datetimelike.py

```diff
@@ -1189,7 +1189,11 @@ def _addsub_object_array(self, other: np.ndarray, op):
         # Caller is responsible for broadcasting if necessary
         assert self.shape == other.shape, (self.shape, other.shape)
 
-        res_values = op(self.astype("O"), np.asarray(other))
+        with warnings.catch_warnings():
```

Comment:
shouln't these not have freq to begin with?

Reply:
`self` here is DTA/TDA/PA, which all have freq.  there is an issue about removing freq from DTA/TDA, which this deprecation is a blocker for